### PR TITLE
Added the README for MoveNet model. Updated the MoveNet constructor s…

### PR DESCRIPTION
**Pull request by yuhuichen1015**
Added the README for the MoveNet model and updated the MoveNet constructor such that it can handle the modelUrl config properly.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)

### CODE REVIEW

**Review by lina128 — CHANGES_REQUESTED** [review]
Looks good overall, I left some minor comments. Thank you Yu-hui! My main concern is the new modelUrl handling logic, see my comments below.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-:-MZL643S2VURMHZ9k9Hx:b-dsatp0)* status: 0 of 1 approvals obtained (waiting on @ardoerlemans, @ronnyvotel, and @yuhuichen1015)

___
*[pose-detection/src/movenet/detector.ts, line 67 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL5nMy0pwhH0V7WetL:-MZL5nMy0pwhH0V7WetM:b-kpzzw0) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/detector.ts#L67)):*
> ```XML
>       this.modelInputResolution.height = MOVENET_SINGLEPOSE_THUNDER_RESOLUTION;
>     } else {
>       if (config.modelUrl.includes("singlepose/lightning")) {
> ```

What is this condition used for? If both modelType and modelUrl can control which model to load, isn't it error prone?
___
*[pose-detection/src/movenet/README.md, line 52 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL3d1k4eRqM5vE2QiY:-MZL3d1k4eRqM5vE2QiZ:b-fsul4h) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/README.md#L52)):*
> ```
> *   *modelUrl* (optional): provide the tf.hub url to download the model. this is
>     used only when the *modelType* is not provided.
> ```
An optional string that specifies custom url of the MoveNet model. This is useful for area/countries that don't have access
to the model hosted on tf.hub.
___
*[pose-detection/src/movenet/README.md, line 65 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL3zQdC-p1e6h3SPxj:-MZL3zQdC-p1e6h3SPxk:b-zijmkt) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/README.md#L65)):*
> ```
> Once the detector is ready, you can start using it to detect poses.
> ```
Now you can use the detector to detect poses.
___
*[pose-detection/src/movenet/README.md, line 76 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL4_6f3Fz2CV2BT205:-MZL4_6f3Fz2CV2BT206:b-fi6eoa) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/README.md#L76)):*
> ```Markdown
> *   *enableSmoothing*: A boolean indicating whether to use temporal filter to
>     smooth the predicted keypoints. Default to *True*.
> ```

The temporal filter relies on the `currentTime` field of the `HTMLVideoElement`. You can override this timestamp by passing in your own timestamp (in microseconds) as the third parameter. This is useful when input is a tensor, which doesn't have the `currentTime` field. Or in testing, to simulate different FPS.

**Review by yuhuichen1015 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)* status: 0 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, and @ronnyvotel)

___
*[pose-detection/src/movenet/detector.ts, line 67 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL5nMy0pwhH0V7WetL:-MZLm-BL4mdy3m58NQpF:b1pzlbo) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/detector.ts#L67)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

What is this condition used for? If both modelType and modelUrl can control which model to load, isn't it error prone?
</blockquote></details>

I'm thinking this is for the users to provide a tf.hub link (e.g. https://tfhub.dev/google/tfjs-model/movenet/singlepose/lightning/2) such that they can specify the version they want, or even for our internal usage to load a model in the x20 folder (which is not pushed to tf.hub yet). In the README, I mentioned that the modelUrl argument is only used when there is no modelType is provided. Do you think that's still confusing? Either way, we need to have a way to specify the modelInputResolution in order to make the model work otherwise the modelUrl argument doesn't work currently. Does that make sense? Thanks
___
*[pose-detection/src/movenet/README.md, line 52 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL3d1k4eRqM5vE2QiY:-MZLndnl89vPJNGECMls:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/README.md#L52)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

> ```
> *   *modelUrl* (optional): provide the tf.hub url to download the model. this is
>     used only when the *modelType* is not provided.
> ```
An optional string that specifies custom url of the MoveNet model. This is useful for area/countries that don't have access
to the model hosted on tf.hub.
</blockquote></details>

Done.
___
*[pose-detection/src/movenet/README.md, line 65 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL3zQdC-p1e6h3SPxj:-MZLnkL5DcG-s2zAmKkC:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/README.md#L65)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

> ```
> Once the detector is ready, you can start using it to detect poses.
> ```
Now you can use the detector to detect poses.
</blockquote></details>

Done.
___
*[pose-detection/src/movenet/README.md, line 76 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL4_6f3Fz2CV2BT205:-MZLo6_10PkMBPpwJKmR:b70dbzj) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/README.md#L76)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

The temporal filter relies on the `currentTime` field of the `HTMLVideoElement`. You can override this timestamp by passing in your own timestamp (in microseconds) as the third parameter. This is useful when input is a tensor, which doesn't have the `currentTime` field. Or in testing, to simulate different FPS.
</blockquote></details>

Thanks! Done

**Review by lina128 — APPROVED** [review]
Thank you Yu-hui!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-:-MZM3QqJ94fV8TpEnJUh:b-7werl2)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, @ronnyvotel, and @yuhuichen1015)

___
*[pose-detection/src/movenet/detector.ts, line 67 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL5nMy0pwhH0V7WetL:-MZM1cqs5DorfunK3C89:bkhk2ic) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/detector.ts#L67)):*
<details><summary><i>Previously, yuhuichen1015 (Yu-Hui Chen) wrote…</i></summary><blockquote>

I'm thinking this is for the users to provide a tf.hub link (e.g. https://tfhub.dev/google/tfjs-model/movenet/singlepose/lightning/2) such that they can specify the version they want, or even for our internal usage to load a model in the x20 folder (which is not pushed to tf.hub yet). In the README, I mentioned that the modelUrl argument is only used when there is no modelType is provided. Do you think that's still confusing? Either way, we need to have a way to specify the modelInputResolution in order to make the model work otherwise the modelUrl argument doesn't work currently. Does that make sense? Thanks
</blockquote></details>

Can you provide a use case when modelType cannot be provided? We can change modelType to be a required field and ask users to always provide one explicitly. Using different versions of the model, either from tf.hub or elsewhere, makes sense. I understand your rationale that this change provide 'best guess' of the resolution in case modelType is not provided, but I just don't think it's a good design to allow both modelUrl and modelType be able to control resolution. Regarding x20, it may not work because of cors.

**Review by ronnyvotel — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, and @yuhuichen1015)

___
*[pose-detection/src/movenet/README.md, line 6 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZNk_xY3KDhC7iuHuo6:-MZNk_xY3KDhC7iuHuo7:b-8azu5i) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5c835a3e5c6b7d887c252192206a0f75eb586ec2/pose-detection/src/movenet/README.md#L6)):*
> ```
> lightning
> ```
thunder?
___
*[pose-detection/src/movenet/README.md, line 76 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZNlGm_6rpO1edRmfQQ:-MZNlGm_6rpO1edRmfQR:b7lgvqy) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5c835a3e5c6b7d887c252192206a0f75eb586ec2/pose-detection/src/movenet/README.md#L76)):*
> ```
> Default
> ```
Defaults
___
*[pose-detection/src/movenet/README.md, line 79 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZNlLC-3iMW_lFTocc2:-MZNlLC-3iMW_lFTocc3:b-iku8sk) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5c835a3e5c6b7d887c252192206a0f75eb586ec2/pose-detection/src/movenet/README.md#L79)):*
> ```
> input
> ```
the input

File: pose-detection/src/movenet/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+## Table of Contents
+
+1.  [Installation](#installation)
+2.  [Usage](#usage)
+3.  [Example Code and Demos](#example-code-and-demos)
+
+## Installation
+
+Please follow the instructions in the Pose API
+[README](https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/README.md)
```
Comment (ardoerlemans) [inline pose-detection/src/movenet/README.md:24]: Can you link to the Installation section here?

https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/README.md#installation

File: pose-detection/src/movenet/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+to install the package.
+
+## Usage
+
+If you are using via npm, you need to import the libraries first.
```
Comment (ardoerlemans) [inline pose-detection/src/movenet/README.md:29]: If you are using the Pose API via npm

File: pose-detection/src/movenet/detector.ts [hunk 3]
```diff
@@ -63,6 +63,14 @@ export class MoveNetDetector extends BasePoseDetector {
     } else if (config.modelType === SINGLEPOSE_THUNDER) {
       this.modelInputResolution.width = MOVENET_SINGLEPOSE_THUNDER_RESOLUTION;
       this.modelInputResolution.height = MOVENET_SINGLEPOSE_THUNDER_RESOLUTION;
+    } else {
+      if (config.modelUrl.includes("singlepose/lightning")) {
```
Comment (ardoerlemans) [inline pose-detection/src/movenet/detector.ts:67]: We could also ask users to supply the model type when they specify a custom URL. But this works for me too.

File: pose-detection/src/movenet/README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+The following code snippet demonstrates how to run the model inference:
+
+```javascript
+const estimationConfig = {enableSmoothing: true};
+const poses = await detector.estimatePoses(image, estimationConfig);
+```
+
+Please refer to the Pose API
+[README](https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/README.md)
```
Comment (ardoerlemans) [inline pose-detection/src/movenet/README.md:90]: Maybe refer to the section that describes the returned poses?

https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/README.md#pose-estimation

File: pose-detection/src/movenet/README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
+# MoveNet
+
+MoveNet is an ultra fast and accurate model that detects 17 keypoints of a body.
+The model is offered on TF.Hub with two variants,
+[Lightning](https://tfhub.dev/google/movenet/singlepose/lightning/3) and
+[Thunder](https://tfhub.dev/google/movenet/singlepose/thunder/3). Lightning is
+intended for latency-critical application, and Thunder is intended for
```
Comment (ardoerlemans) [inline pose-detection/src/movenet/README.md:7]: latency-critical applications

**Review by ardoerlemans — APPROVED** [review]
Thanks for adding this! Only a few minor suggestions.

**Review by yuhuichen1015 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, @ronnyvotel, and @yuhuichen1015)

___
*[pose-detection/src/movenet/detector.ts, line 67 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL5nMy0pwhH0V7WetL:-MZO57dP29Q11N-pv2bE:b-fuu1ld) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/detector.ts#L67)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Can you provide a use case when modelType cannot be provided? We can change modelType to be a required field and ask users to always provide one explicitly. Using different versions of the model, either from tf.hub or elsewhere, makes sense. I understand your rationale that this change provide 'best guess' of the resolution in case modelType is not provided, but I just don't think it's a good design to allow both modelUrl and modelType be able to control resolution. Regarding x20, it may not work because of cors.
</blockquote></details>

Sorry I'm a bit confused. I thought the modelUrl argument was added by you (or Ard)? If we don't add those logics, the modelUrl argument won't work since the code doesn't know and set the input resolution when people provide the modelUrl.

I agree that guessing the resolution from the model name isn't the best solution and might cause some issues. How about we let the users to provide the input resolution when they specify the modelUrl?

Regarding the use case, we have actually been using the modelUrl argument internally when testing out our models in x20 before we actually push to tf.hub. I think it'll be convenient to keep that argument for testing purposes. The proposed logics are like:

1) Check modelType, if it is not null, then use modelType to load the model from tf.hub.
2) If modelType is null, look for modelUrl to load the model. If modelUrl is provided, the inputWidth/inputHeight need to also be provided by the users.

WDYT?
___
*[pose-detection/src/movenet/README.md, line 6 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZNk_xY3KDhC7iuHuo6:-MZO79IM-atBIF-rXZfC:b-r8ka10) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5c835a3e5c6b7d887c252192206a0f75eb586ec2/pose-detection/src/movenet/README.md#L6)):*
<details><summary><i>Previously, ronnyvotel wrote…</i></summary><blockquote>

> ```
> lightning
> ```
thunder?
</blockquote></details>

Ah thanks for the catch.
___
*[pose-detection/src/movenet/README.md, line 76 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZNlGm_6rpO1edRmfQQ:-MZO7FTE-yD51eRi-Pi2:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5c835a3e5c6b7d887c252192206a0f75eb586ec2/pose-detection/src/movenet/README.md#L76)):*
<details><summary><i>Previously, ronnyvotel wrote…</i></summary><blockquote>

> ```
> Default
> ```
Defaults
</blockquote></details>

Done.
___
*[pose-detection/src/movenet/README.md, line 79 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZNlLC-3iMW_lFTocc2:-MZO7LCk1v_CcJPj6Igd:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5c835a3e5c6b7d887c252192206a0f75eb586ec2/pose-detection/src/movenet/README.md#L79)):*
<details><summary><i>Previously, ronnyvotel wrote…</i></summary><blockquote>

> ```
> input
> ```
the input
</blockquote></details>

Done.
___
*[pose-detection/src/movenet/README.md, line 7 at r3](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZKqJcAOFlJH-k6njQS-r3-7:-MZOP3wE5BcrCl38_Cm2:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/755e8a8a0e87003e41944a200a47619add851cce/pose-detection/src/movenet/README.md#L7)):*
<details><summary><i>Previously, ardoerlemans (Ard Oerlemans) wrote…</i></summary><blockquote>

latency-critical applications
</blockquote></details>

Done.
___
*[pose-detection/src/movenet/README.md, line 24 at r3](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZKqJcAOFlJH-k6njQS-r3-24:-MZOPCek0YlESqTabK-8:bglyovg) ([raw file](https://github.com/tensorflow/tfjs-models/blob/755e8a8a0e87003e41944a200a47619add851cce/pose-detection/src/movenet/README.md#L24)):*
<details><summary><i>Previously, ardoerlemans (Ard Oerlemans) wrote…</i></summary><blockquote>

Can you link to the Installation section here?

https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/README.md#installation
</blockquote></details>

Thanks for the suggestion. Done.
___
*[pose-detection/src/movenet/README.md, line 90 at r3](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZKqJcAOFlJH-k6njQS-r3-90:-MZOPPMY8cDDPe_ZAnN8:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/755e8a8a0e87003e41944a200a47619add851cce/pose-detection/src/movenet/README.md#L90)):*
<details><summary><i>Previously, ardoerlemans (Ard Oerlemans) wrote…</i></summary><blockquote>

Maybe refer to the section that describes the returned poses?

https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/README.md#pose-estimation
</blockquote></details>

Done.

**Review by yuhuichen1015 — COMMENTED** [review]
Thanks for all of your suggestions! I think we have only one issue left to address regarding the modelUrl.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-:-MZOPwEK1rE85ZSGLq5P:b-bo6q4y)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, @ronnyvotel, and @yuhuichen1015)

**Review by yuhuichen1015 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, @ronnyvotel, and @yuhuichen1015)

___
*[pose-detection/src/movenet/README.md, line 29 at r3](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZKqJcAOFlJH-k6njQS-r3-29:-MZOfqx8DSInkdgkTFGX:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/755e8a8a0e87003e41944a200a47619add851cce/pose-detection/src/movenet/README.md#L29)):*
<details><summary><i>Previously, ardoerlemans (Ard Oerlemans) wrote…</i></summary><blockquote>

If you are using the Pose API via npm
</blockquote></details>

Done.

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, @ronnyvotel, and @yuhuichen1015)

___
*[pose-detection/src/movenet/detector.ts, line 67 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL5nMy0pwhH0V7WetL:-MZOkF3kDiKR-VtroqDB:b-elky7d) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/detector.ts#L67)):*
<details><summary><i>Previously, yuhuichen1015 (Yu-Hui Chen) wrote…</i></summary><blockquote>

Sorry I'm a bit confused. I thought the modelUrl argument was added by you (or Ard)? If we don't add those logics, the modelUrl argument won't work since the code doesn't know and set the input resolution when people provide the modelUrl.

I agree that guessing the resolution from the model name isn't the best solution and might cause some issues. How about we let the users to provide the input resolution when they specify the modelUrl?

Regarding the use case, we have actually been using the modelUrl argument internally when testing out our models in x20 before we actually push to tf.hub. I think it'll be convenient to keep that argument for testing purposes. The proposed logics are like:

1) Check modelType, if it is not null, then use modelType to load the model from tf.hub.
2) If modelType is null, look for modelUrl to load the model. If modelUrl is provided, the inputWidth/inputHeight need to also be provided by the users.

WDYT?
</blockquote></details>

Haha, I'm confused too. Yes, I added modelUrl argument, but it is only supposed to provide url to load the model, in my mind, the modelType is separate, it controls the resolution. What you described here could definitely work, but isn't it a complex logic? How do you communicate this logic to user? e.g. When should they provide modelType and when not? What naming convention should they follow for the modelUrl?

**Review by yuhuichen1015 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, and @ronnyvotel)

___
*[pose-detection/src/movenet/detector.ts, line 67 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZL5nMy0pwhH0V7WetL:-MZPDt601bTWpN8TwRL0:butqkao) ([raw file](https://github.com/tensorflow/tfjs-models/blob/b249967bc21caa8e43905907684cb54843d9d6c7/pose-detection/src/movenet/detector.ts#L67)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Haha, I'm confused too. Yes, I added modelUrl argument, but it is only supposed to provide url to load the model, in my mind, the modelType is separate, it controls the resolution. What you described here could definitely work, but isn't it a complex logic? How do you communicate this logic to user? e.g. When should they provide modelType and when not? What naming convention should they follow for the modelUrl?
</blockquote></details>

Sorry for the confusion! Synced offline and I think we are on the same page that the users should either:

1) Provide only "modelType", which will load the model from tf.hub or
2) Provide BOTH "modelType" and "modelUrl". In which case, the input resolution is determined by modelType and the detector will load the model from modelUrl.

Reverted the logics here and updated the README. PTAL.
___
*[pose-detection/src/movenet/detector.ts, line 67 at r3](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-MZKqJcAOFlJH-k6njQR-r3-67:-MZPDlFC8zwYr-y8hi70:bpa6uqw) ([raw file](https://github.com/tensorflow/tfjs-models/blob/755e8a8a0e87003e41944a200a47619add851cce/pose-detection/src/movenet/detector.ts#L67)):*
<details><summary><i>Previously, ardoerlemans (Ard Oerlemans) wrote…</i></summary><blockquote>

We could also ask users to supply the model type when they specify a custom URL. But this works for me too.
</blockquote></details>

Yeah I think that's a good idea and is actually how it was designed to use after chatting with Na. Removed the logics and updated the description.

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-:-MZPI1Vu2Ql32iDp6CV9:bb6ywmx)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans, @lina128, and @ronnyvotel)

**Review by ronnyvotel — APPROVED** [review]
> Reviewed 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678)* status: :shipit: complete! 3 of 1 approvals obtained (waiting on @ardoerlemans and @lina128)

**Review by ronnyvotel — APPROVED** [review]
:lgtm_strong:

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/678#-:-MZPYBSz9Vusf3FURtXg:b-jbsf8z)* status: :shipit: complete! 3 of 1 approvals obtained (waiting on @ardoerlemans and @lina128)

**Pull request merged**